corehq/apps/app_manager/models.py

```diff
@@ -792,6 +792,43 @@ def short_comment(self):
         return self.comment if len(self.comment) <= 500 else self.comment[:497] + '...'
 
 
+class CaseLoadReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    path = StringProperty()
+    properties = ListProperty(unicode)
+
+
+class CaseSaveReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    case_type = StringProperty()
+    path = StringProperty()
+    properties = ListProperty(unicode)
+    create = BooleanProperty(default=False)
+    close = BooleanProperty(default=False)
+
+
+class CaseReferences(DocumentSchema):
+    _allow_dynamic_properties = False
+    load = DictProperty()
+    save = DictProperty()
```

Comment:
I think having this validated is great. If there were a bug in Vellum that produced the wrong format then we'd end up needing to do a data migration to fix it which seems like a more difficult situation to fix.